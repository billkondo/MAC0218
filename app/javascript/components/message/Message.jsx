import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import Snackbar from '@material-ui/core/Snackbar';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

const styles = theme => ({
  close: {
    padding: theme.spacing.unit / 2,
  },
});

class Message extends React.Component {

  render() {
    const { classes } = this.props;
    return (
      <div>
        <Snackbar
          anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'left',
          }}
          open={this.props.open}
          autoHideDuration={6000}
          onClose={() => this.props.toggleMenu()}
          ContentProps={{
            'aria-describedby': 'message-id',
          }}
          message={<span id="message-id">{this.props.mensagem}</span>}
          action={[
            <IconButton
              key="close"
              aria-label="Close"
              color="inherit"
              className={classes.close}
              onClick={() => this.props.toggleMenu()}
            >
              <CloseIcon />
            </IconButton>,
          ]}
        />
      </div>
    );
  }
}

Message.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(Message);