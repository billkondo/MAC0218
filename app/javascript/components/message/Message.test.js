import React from 'react'
import  Message from './Message'
import { mount, configure } from 'enzyme'
import toJson from "enzyme-to-json"
import Adapter from 'enzyme-adapter-react-16'
import { BrowserRouter as Router } from 'react-router-dom';

configure({adapter: new Adapter()});
it("should render the Message component correctly", () => {
  const app = mount(<Message />)
  expect(toJson(app)).toMatchSnapshot()
  app.unmount
})
