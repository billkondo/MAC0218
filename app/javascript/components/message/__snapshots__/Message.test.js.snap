// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Message component correctly 1`] = `
<WithStyles(Message)>
  <Message
    classes={
      Object {
        "close": "Message-close-1",
      }
    }
  >
    <div>
      <WithStyles(ForwardRef(Snackbar))
        ContentProps={
          Object {
            "aria-describedby": "message-id",
          }
        }
        action={
          Array [
            <ForwardRef(WithStyles)
              aria-label="Close"
              className="Message-close-1"
              color="inherit"
              onClick={[Function]}
            >
              <pure(CloseIcon) />
            </ForwardRef(WithStyles)>,
          ]
        }
        anchorOrigin={
          Object {
            "horizontal": "left",
            "vertical": "bottom",
          }
        }
        autoHideDuration={6000}
        message={
          <span
            id="message-id"
          />
        }
        onClose={[Function]}
      >
        <ForwardRef(Snackbar)
          ContentProps={
            Object {
              "aria-describedby": "message-id",
            }
          }
          TransitionComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "Naked": Object {
                "$$typeof": Symbol(react.forward_ref),
                "muiSupportAuto": true,
                "propTypes": Object {
                  "children": [Function],
                  "in": [Function],
                  "onEnter": [Function],
                  "onExit": [Function],
                  "style": [Function],
                  "theme": [Function],
                  "timeout": [Function],
                },
                "render": [Function],
              },
              "displayName": "WithTheme(ForwardRef(Grow))",
              "muiSupportAuto": true,
              "propTypes": Object {
                "innerRef": [Function],
              },
              "render": [Function],
            }
          }
          action={
            Array [
              <ForwardRef(WithStyles)
                aria-label="Close"
                className="Message-close-1"
                color="inherit"
                onClick={[Function]}
              >
                <pure(CloseIcon) />
              </ForwardRef(WithStyles)>,
            ]
          }
          anchorOrigin={
            Object {
              "horizontal": "left",
              "vertical": "bottom",
            }
          }
          autoHideDuration={6000}
          classes={
            Object {
              "anchorOriginBottomCenter": "MuiSnackbar-anchorOriginBottomCenter",
              "anchorOriginBottomLeft": "MuiSnackbar-anchorOriginBottomLeft",
              "anchorOriginBottomRight": "MuiSnackbar-anchorOriginBottomRight",
              "anchorOriginTopCenter": "MuiSnackbar-anchorOriginTopCenter",
              "anchorOriginTopLeft": "MuiSnackbar-anchorOriginTopLeft",
              "anchorOriginTopRight": "MuiSnackbar-anchorOriginTopRight",
              "root": "MuiSnackbar-root",
            }
          }
          disableWindowBlurListener={false}
          message={
            <span
              id="message-id"
            />
          }
          onClose={[Function]}
          transitionDuration={
            Object {
              "enter": 225,
              "exit": 195,
            }
          }
        />
      </WithStyles(ForwardRef(Snackbar))>
    </div>
  </Message>
</WithStyles(Message)>
`;
